include/aws/common/zero.h

```diff
@@ -0,0 +1,99 @@
+#ifndef AWS_COMMON_ZERO_H
+#define AWS_COMMON_ZERO_H
+
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/stdbool.h>
+#include <aws/common/stdint.h>
+
+#include <string.h>
+
+/**
+ * Set each byte in the struct to zero.
+ */
+#define AWS_ZERO_STRUCT(object)                                                                                        \
+    do {                                                                                                               \
+        memset(&(object), 0, sizeof(object));                                                                          \
+    } while (0)
+
+/**
+ * Set each byte in the array to zero.
+ * Does not work with arrays of unknown bound.
+ */
+#define AWS_ZERO_ARRAY(array)                                                                                          \
+    do {                                                                                                               \
+        memset((void *)(array), 0, sizeof(array));                                                                     \
+    } while (0)
+
+/**
+ * Returns whether each byte in the object is zero.
+ */
+#ifdef CBMC
+/* clang-format off */
+#    define AWS_IS_ZEROED(object)                                                                                      \
+        __CPROVER_forall {                                                                                             \
+            int i;                                                                                                     \
+            (i >= 0 && i < sizeof(object)) ==> ((const uint8_t *)&object)[i] == 0                                      \
+        }
+/* clang-format on */
+#else
+#    define AWS_IS_ZEROED(object) aws_is_zeroed(&(object), sizeof(object))
+#endif
+
+/**
+ * Returns whether each byte is zero.
+ */
+AWS_STATIC_IMPL
```

Comment:
make sure codegen.c includes this file plz.

Reply:
it does. common.h includes zero.h